File: 03417. Zigzag Grid Traversal With Skip/solution.cs
/**
 * Problem: 3417. Zigzag Grid Traversal With Skip
 *
 * Difficulty: Easy
 *
 * Language: C#
 *
 * Performance: Runtime - 0 ms (Beats 100%)
 */

public class Solution
{
  public IList<int> ZigzagTraversal(int[][] grid)
  {
    // Get number of rows and columns in the grid
    int rows = grid.Length,
      cols = grid[0].Length;

    // List to store collected elements
    var result = new List<int>();

    // Iterate through each row in the grid
    for (int row = 0; row < rows; row++)
    {
      // Determine starting column based on row parity
      // Even row: start from column 0 (left)
      // Odd row: start from last column or second last based on column count
      int startCol = row % 2 == 0 ? 0 : (cols % 2 == 0 ? cols - 1 : cols - 2);

      // Determine step direction
      // Even row: move right (step +2)
      // Odd row: move left (step -2)
      int step = row % 2 == 0 ? 2 : -2;

      // Traverse columns with step of 2
      for (int col = startCol; col >= 0 && col < cols; col += step)
        result.Add(grid[row][col]);
    }

    // Return the final result
    return result;
  }
}

File: 03417. Zigzag Grid Traversal With Skip/solution.go
/**
 * Problem: 3417. Zigzag Grid Traversal With Skip
 *
 * Difficulty: Easy
 *
 * Language: Golang
 *
 * Performance: Runtime - 0 ms (Beats 100%)
 */

func zigzagTraversal(grid [][]int) []int {
	// Slice to store collected elements
	result := make([]int, 0)

	// Iterate through each row in the grid
	for rowIndex := 0; rowIndex < len(grid); rowIndex++ {
		// If row index is even, traverse left to right
		if rowIndex%2 == 0 {
			// Iterate through columns from left to right
			for colIndex := 0; colIndex < len(grid[0]); colIndex++ {
				// Add element if sum of indices is even
				if (rowIndex+colIndex)%2 == 0 {
					result = append(result, grid[rowIndex][colIndex])
				}
			}
		} else {
			// If row index is odd, traverse right to left

			// Iterate through columns from right to left
			for colIndex := len(grid[0]) - 1; colIndex >= 0; colIndex-- {
				// Add element if sum of indices is even
				if (rowIndex+colIndex)%2 == 0 {
					result = append(result, grid[rowIndex][colIndex])
				}
			}
		}
	}

	// Return the final result
	return result
}
